src/apify_client/clients/resource_clients/run.py

```diff
@@ -0,0 +1,169 @@
+        Args:
+            wait_secs (int, optional): how long does the client wait for run to finish. None for indefinite.
+
+        Returns:
+            dict, optional: The actor run data. If the status on the object is not one of the terminal statuses
+                (SUCEEDED, FAILED, TIMED_OUT, ABORTED), then the run has not yet finished.
+        """
+        return self._wait_for_finish(wait_secs=wait_secs)
+
+    def metamorph(
+        self,
+        *,
+        target_actor_id: str,
+        target_actor_build: Optional[str] = None,
+        run_input: Optional[Any] = None,
+        content_type: Optional[str] = None,
+    ) -> Dict:
+        """Transform an actor run into a run of another actor with a new input.
+
+        https://docs.apify.com/api/v2#/reference/actor-runs/metamorph-run/metamorph-run
+
+        Args:
+            target_actor_id (str): ID of the target actor that the run should be transformed into
+            target_actor_build (str, optional): The build of the target actor. It can be either a build tag or build number.
+                By default, the run uses the build specified in the default run configuration for the target actor (typically the latest build).
+            run_input (Any): The input to pass to the new run.
+            content_type (str, optional): The content type of the input.
+
+        Returns:
+            dict: The actor run data.
+        """
+        run_input, content_type = _encode_key_value_store_record_value(run_input, content_type)
+
+        safe_target_actor_id = _to_safe_id(target_actor_id)
+
+        request_params = self._params(
+            targetActorId=safe_target_actor_id,
+            build=target_actor_build,
+        )
+
+        response = self.http_client.call(
+            url=self._url('metamorph'),
+            method='POST',
+            headers={'content-type': content_type},
+            data=run_input,
+            params=request_params,
+        )
+
+        return _parse_date_fields(_pluck_data(response.json()))
+
+    def resurrect(self) -> Dict:
+        """Resurrect a finished actor run.
+
+        Only finished runs, i.e. runs with status FINISHED, FAILED, ABORTED and TIMED-OUT can be resurrected.
+        Run status will be updated to RUNNING and its container will be restarted with the same default storages.
+
+        https://docs.apify.com/api/v2#/reference/actor-runs/resurrect-run/resurrect-run
+
+        Returns:
+            dict: The actor run data.
+        """
+        response = self.http_client.call(
+            url=self._url('resurrect'),
+            method='POST',
+            params=self._params(),
+        )
+
+        return _parse_date_fields(_pluck_data(response.json()))
+
+    def dataset(self) -> DatasetClient:
+        """Get the client for the default dataset of the actor run.
+
+        Currently this works only through `actor_client.last_run().dataset()`.
+        It will become available for all runs once the API supports it.
```

Comment:
I think that we should go around the API and always allow this function to work. We have the dataset id, so it should be possible. It would be very bad not to support this. If you want to keep it working like this, at least move it to the ActorClient and make it something like get_last_run_dataset(), so it doesn't confuse people.
Same for the key_value_store(), request_queue() and log() functions.

Reply:
The endpoints necessary for this to work with a generic run, rather than just the last one, are actually just around the corner, they're being worked on right now, so I'll keep it as it is for now, and then just remove the comments once the new endpoints are deployed to the API.